administration/src/application/components/forms/RegionForm.tsx

```diff
@@ -0,0 +1,69 @@
+import { Region, ShortTextInput, useGetRegionByPostalCodeQuery } from '../../../generated/graphql'
+import { useUpdateStateCallback } from '../../useUpdateStateCallback'
+import { Form } from '../../FormType'
+import {
+  CompoundState,
+  createCompoundGetArrayBufferKeys,
+  createCompoundValidate,
+  createCompoundInitialState,
+} from '../../compoundFormUtils'
+import SelectForm, { SelectItem } from '../primitive-inputs/SelectForm'
+import { ProjectConfigContext } from '../../../project-configs/ProjectConfigContext'
+import { useContext } from 'react'
+import { useSnackbar } from 'notistack'
+
+const SubForms = {
+  region: SelectForm,
+}
+
+type State = CompoundState<typeof SubForms>
+type ValidatedInput = { region: ShortTextInput }
+type Options = {}
+type AdditionalProps = { regionData: Region[]; postalCode: string }
+
+const getOptionsLabel = (prefix: string, name: string) => `${name} (${prefix})`
+
+export const getOptions = (regions: Region[]): SelectItem[] =>
+  regions.map(region => {
+    return { label: getOptionsLabel(region.prefix, region.name), value: region.id.toString() }
+  })
+
+const RegionForm: Form<State, Options, ValidatedInput, AdditionalProps> = {
+  initialState: { ...createCompoundInitialState(SubForms) },
+  getArrayBufferKeys: createCompoundGetArrayBufferKeys(SubForms),
+  validate: createCompoundValidate(SubForms, { region: { items: [] } }),
```

Comment:
i'm not sure how to provide proper validation data, since i get the regionData that i need via additional props and not hardcoded as in `OrganizationForm` i even would know why we can't deactivate the validation for region according to options provided

Reply:
As a rule of thumb: All the data that you need for validation (and that you cannot hardcode) should go into the `Options` slot, not in the `AdditionalProps` slot. Additional props are conceptually for styling rather than content stuff^^
You can access options both in the component (via the `options` prop) and inside the validate function (via the second argument).

We can also go over this together if you want :)